### ISSUE TITLE
Improve error reporting in palette function

### ISSUE THREAD

**Issue by fosterinteractive** [fosterinteractive/mainspring#28]
It would be better to report where the function is called from if a call is made to a non-existant color map value. 

_g-color__functions.scss
``` 
Error: scss/global/g-color/_g-color__functions.scss
Error: xxdark is not a valid key in palettes
        on line 47 of scss/global/g-color/_g-color__functions.scss
>>     @error $tone + ' is not a valid key in palettes';
```

at-import's respond-to mixin does this when you call a non-existant breakpoint. Perhaps the same approach can be used to make the error message more useful.

**Comment by pierremarcelio** [comment]
Yes that would be awesome, I spent a lot time fixing those missing palettes. Let me know if left any orphan colour on my last commit.
